### PR TITLE
all: Renamed all .h and .c to .hpp and .cpp

### PR DESCRIPTION
**Pull request by sjalander**
Renamed all .c and .h files to .cpp and .hpp since it's relally C++ code.

### CODE REVIEW

File: tests/rvsdg/traverser/test-cone-traverser.cpp [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
-#include <jive/view.h>
-#include <jive/vsdg.h>
+#include <jive/view.hpp>
+#include <jive/vsdg.hpp>
 
 #include "testnodes.h"
 
 typedef struct graph_desc {
-	std::unique_ptr<jive::graph> graph;
+	std::unique_ptr<jive::graphpp> graph;
```
Comment (phate) [inline tests/rvsdg/traverser/test-cone-traverser.cpp]: Why did the name change here?

File: tests/rvsdg/traverser/test-cone-traverser.cpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ static graph_desc
 prepare_graph()
 {
 	graph_desc g;
-	g.graph = std::move(std::unique_ptr<jive::graph>(new jive::graph()));
+	g.graph = std::move(std::unique_ptr<jive::graphpp>(new jive::graph()));
```
Comment (phate) [inline tests/rvsdg/traverser/test-cone-traverser.cpp]: The same here, where does all in a sudden the name change come from?

**Comment by phate** [comment]
You changed none of the file guards. If you rename the files, the guards have to change as well.

**Comment by sjalander** [comment]
The issues have been address.
Checks (jive and jlm) and polybench still work.

**Pull request merged**